lib/core/client.js

```diff
@@ -506,7 +477,7 @@ class Parser extends HTTPParser {
     // have no way of slicing the parsing buffer without knowing
     // the offset which is only provided in kOnExecute.
     if (upgrade) {
-      const head = this.getCurrentBuffer().slice(ret)
+      const head = currentBuffer.slice(ret)
```

Comment:
I guess you could add `getCurrentBuffer()` to the parser and then we don't need to change anything here.

Reply:
I think it should be the `data` passed to `execute` though, not the uint8 typed array.